src/pkg/artifact/manager.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifact
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"time"
+)
+
+var (
+	// Mgr is a global artifact manager instance
+	Mgr = NewManager()
+)
+
+// Manager is the only interface of artifact module to provide the
+// management functions for artifacts
+type Manager interface {
+	// List artifacts according to the query
+	List(ctx context.Context, query ...*q.Query) (total int64, artifacts []*Artifact, err error)
+	// Get the artifact specified by the ID
+	Get(ctx context.Context, id int64) (*Artifact, error)
+	// Create the artifact specified by the digest under the repository,
+	// returns the artifact ID and error. If tags are provided, attach them
+	// to the artifact. If the tag is already attached to other artifact,
+	// will detach it first
+	Create(ctx context.Context, repository *models.RepoRecord, digest string, tags ...string) (id int64, err error)
+	// Delete just deletes the artifact record. The underlying data of registry will be
+	// removed during garbage collection
+	Delete(ctx context.Context, id int64) error
+	// UpdatePullTime updates the pull time of artifact. If the tag is provided, update the pull time of tag as well
+	UpdatePullTime(ctx context.Context, artifact *Artifact, tag *Tag, time time.Time) error
+	// GetSubResource returns the sub resource of the artifact
+	// The sub resource is different according to the artifact type:
+	// build history for image; values.yaml, readme and dependencies for chart, etc
+	GetSubResource(ctx context.Context, artifactID int64, resource string) ([]byte, error)
```

Comment:
How about

```go
GetSubResources(ctx context.Context, artifactID int64, resources ...string) (map[string][]byte, error)
```

In case if I want to get them all or more than one?
?

Reply:
ok, keep `GetSubResource ` as it is